### ISSUE TITLE
Coroutine by Tween.ToYieldInstruction() may wait for other Tween animations

### ISSUE THREAD

**Issue by kyubuns** [KyryloKuzyk/PrimeTween#126]
I'm sorry I couldn't write the minimum reproducible code.
This occurred randomly, and even if I tried to write a specific reproduction code, it would not reproduce.

When I tried it with the debugger, when I was waiting for the Coroutine I got from TweenA, and when TweenCoroutineEnumerator.resetEnumerator and TweenCoroutineEnumerator.SetTween were called within a same frame, the Coroutine I got from TweenA sometimes waited for another Tween.
If these two are called in the same frame, it seems that TweenCoroutineEnumerator.IEnumerator.MoveNext will return true by mistake.

Is it my misunderstanding, or is this possible in the PrimeTweenManager Pool?

**Comment by KyryloKuzyk** [comment]
Thank you for opening the issue!

Hmm, PrimeTween has several automated tests that should cover exactly this behavior: YieldInstructionsClash2() and YieldInstructionsClash(). 

Do you experience this problem only when running with the debugger attached?

**Comment by kyubuns** [comment]
> Do you experience this problem only when running with the debugger attached?

No, it can happen at any time.

I was unable to generate this phenomenon for a single Tween, but I was able to write code that could generate it randomly.

```csharp
using System.Collections;
using PrimeTween;
using UnityEngine;

public class Sample : MonoBehaviour
{
    public void Start()
    {
        StartCoroutine(Coroutine());
    }

    public void Update()
    {
        StartCoroutine(Coroutine());
    }

    IEnumerator Coroutine()
    {
        var testObject = new GameObject("Test");
        var tween = Tween.PositionX(testObject.transform, endValue: 10, duration: 1);
        var startTime = Time.time;
        yield return tween.ToYieldInstruction();
        Debug.Log($"Finish {Time.time - startTime}");
    }
}
```

- Unity version: Unity6000.0.15f1 and 6000.0.22f1
- PrimeTween version: 1.2.0

I expect the debug log to output a value close to 1.0.
But, it will give a value close to 1 or 2.

![CleanShot_20241121-205903@2x](https://github.com/user-attachments/assets/eea7ae59-6520-4102-8684-a40c610ec545)

**Comment by KyryloKuzyk** [comment]
Huge thank you for the sample! I managed to reproduce the issue on my side and already working on a fix.

**Comment by KyryloKuzyk** [comment]
I fixed the issue in version 1.2.1. Huge thank you for the bug report!
https://github.com/KyryloKuzyk/PrimeTween/blob/main/changelog.md
